pkg/odo/cli/service/create.go

```diff
@@ -190,6 +210,31 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 		return nil
 	}
 
+	// we want to find an Operator only if something's passed to the crd flag on CLI
+	if experimental.IsExperimentalModeEnabled() && o.Crd != "" {
+		// make sure that CSV of the specified ServiceType exists
+		csv, err := o.KClient.GetClusterServiceVersion(o.ServiceType)
+		if err != nil {
+			// error only occurs when OperatorHub is not installed.
+			// k8s does't have it installed by default but OCP does
+			return err
+		}
+
+		var almExamples []map[string]interface{}
+		err = json.Unmarshal([]byte(csv.Annotations["alm-examples"]), &almExamples)
+		if err != nil {
+			return errors.Wrap(err, "unable to unmarshal alm-examples")
+		}
+
+		for _, example := range almExamples {
+			if example["kind"].(string) == o.Crd {
```

Comment:
if the Crd is doesn't match here then odo will assume it as a service catalog request, so `odo service create  etcdOperator --crdEtdCluster --dry-run` would not show error and instead try to create service catalog stuff If i am not wrong.

Reply:
Since you brought up similar point in a different PR (https://github.com/openshift/odo/pull/2739#pullrequestreview-387155338), this has been addressed.